nextstrain/cli/util.py

```diff
@@ -68,9 +75,36 @@ def check_for_new_version():
         else:
             python = sys.executable
     else:
-        python = next(filter(which, ["python3", "python"])) or "python3"
+        python = next(filter(which, ["python3", "python"]), "python3")
+
+    # Find our installer (e.g. pip, conda).
+    try:
+        distribution = distribution_info("nextstrain-cli")
+    except PackageNotFoundError:
+        installer = None
+    else:
+        installer = (distribution.read_text("INSTALLER") or '').strip() or None
+
+    # Determine if we're pipx or not.
+    if installer == "pip" and "/pipx/venvs/nextstrain-cli/" in python:
+        installer = "pipx"
+
+    # Find our Conda details, if applicable.
+    if installer == "conda":
+        # Prefer mamba over conda, if available on PATH.
+        conda = next(filter(which, ["mamba", "conda"]), "mamba")
```

Comment:
Good question!

If we can't find `mamba` or `conda` on the current `PATH` but we're positively installed via a Conda package, then we have to assume that a) one of them is available _somehow_ and b) that the user knows (or will figure out) how to run one of them. We could default to either `conda` or `mamba`, but since our official install instructions use `mamba` and `conda` sometimes has dep-solving memory issues, I figured that `mamba` was the best thing to use in this edge case.

There's a few reasons why neither command may be available on `PATH` when we go looking with `which`, e.g. the commands could be off `PATH` but available via shell alias or shell function wrappers or they could be behind a modules system like Environment Modules and need enabling with e.g. `module load …`.

Reply:
Repushed 1a1c6f7 with this explanation as a comment.